pkg/scyllaclient/client_agent.go

```
@@ -262,3 +262,25 @@ func (c *Client) FreeOSMemory(ctx context.Context, host string) error {
 	_, err := c.agentOps.FreeOSMemory(&p)
 	return errors.Wrap(err, "free OS memory")
 }
+
+// CloudMetadata returns instance metadata from agent node.
+func (c *Client) CloudMetadata(ctx context.Context, host string) (InstanceMetadata, error) {
+	p := operations.MetadataParams{
+		Context: forceHost(ctx, host),
+	}
+
+	meta, err := c.agentOps.Metadata(&p)
+	if err != nil {
+		return InstanceMetadata{}, errors.Wrap(err, "cloud metadata")
+	}
+
+	payload := meta.GetPayload()
+	if payload == nil {
+		return InstanceMetadata{}, nil
+	}
```

I would say that the error checking should be enough (it should be reasonable to assume that agent does not hide errors), but if we are also validating that `payload != nil`, then we should return an error if the validation fails.

---

I'm scared of nil pointer dereference panic, so I've changed this code to return error :smile: